classes/migration/upgrade/v3_4_0/I7434_WorkType.inc.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/**
+ * @file classes/migration/upgrade/v3_4_0/I7434_WorkType.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class I7434_WorkType
+ * @brief Consolidate submission workType settings in the submissions table
+ */
+
+namespace APP\migration\upgrade\v3_4_0;
+
+use Illuminate\Support\Facades\DB;
+use PKP\config\Config;
+
+class I7434_WorkType extends \PKP\migration\Migration
+{
+    /**
+     * Run the migration.
+     */
+    public function up(): void
+    {
+        // PostgreSQL and MySQL use different type casts for integers
+        $type = 'UNSIGNED';
+        if (substr(Config::getVar('database', 'driver'), 0, strlen('postgres')) === 'postgres') {
+            $type = 'INTEGER';
+        }
+        DB::statement(
+            'UPDATE submissions as s, submission_settings as ss
+            SET s.work_type = CAST(ss.setting_value as ' . $type . ")
+            WHERE ss.submission_id = s.submission_id
+                AND ss.setting_name = 'workType'"
+        );
```

Comment:
@asmecher can I get an assist here? This is working for MySQL but in Postgres I get the following error:

```
ERROR: SQLSTATE[42601]: Syntax error: 7 ERROR:  syntax error at or near ","
LINE 1: UPDATE submissions as s, submission_settings as ss
                               ^ (SQL: UPDATE submissions as s, submission_settings as ss
            SET s.work_type = CAST(ss.setting_value as INTEGER)
            WHERE ss.submission_id = s.submission_id
                AND ss.setting_name = 'workType')
```

Reply:
Sorry for the wait! The equivalent for PostgreSQL would be...
```sql
UPDATE submissions SET work_type = CAST(ss.setting_value AS INTEGER) FROM submission_settings AS ss
WHERE ss.submission_id = submissions.submission_id
AND ss.setting_name = 'workType';
```